plugin/src/helpers/functions.ts

```diff
@@ -34,30 +34,39 @@ const writeApiFunction = async ({ appDir, functionDir }) => {
 export const writeFunctions = async ({
   constants,
   netlifyConfig,
+  neededFunctions,
 }: {
   constants: NetlifyPluginConstants
   netlifyConfig: NetlifyConfig
+  neededFunctions: Array<string>
 }): Promise<void> => {
   const { PUBLISH_DIR, INTERNAL_FUNCTIONS_SRC } = constants
   const siteRoot = getGatsbyRoot(PUBLISH_DIR)
   const functionDir = resolve(INTERNAL_FUNCTIONS_SRC, '__api')
   const appDir = relative(functionDir, siteRoot)
 
-  await writeFunction({
-    renderMode: 'SSR',
-    handlerName: '__ssr',
-    appDir,
-    functionsSrc: INTERNAL_FUNCTIONS_SRC,
-  })
+  if (neededFunctions.includes('SSR')) {
+    await writeFunction({
+      renderMode: 'SSR',
+      handlerName: '__ssr',
+      appDir,
+      functionsSrc: INTERNAL_FUNCTIONS_SRC,
+    })
+  }
+
+  if (neededFunctions.includes('DSG')) {
+    await writeFunction({
+      renderMode: 'DSG',
+      handlerName: '__dsg',
+      appDir,
+      functionsSrc: INTERNAL_FUNCTIONS_SRC,
+    })
+  }
 
-  await writeFunction({
-    renderMode: 'DSG',
-    handlerName: '__dsg',
-    appDir,
-    functionsSrc: INTERNAL_FUNCTIONS_SRC,
-  })
-  await setupImageCdn({ constants, netlifyConfig })
-  await writeApiFunction({ appDir, functionDir })
+  if (neededFunctions.includes('API')) {
+    await setupImageCdn({ constants, netlifyConfig })
```

Comment:
Should this be inside this conditional?

Reply:
Currently, if `NETLIFY_SKIP_GATSBY_FUNCTIONS === true` then this gets skipped. Do we want to keep that functionality and the enable it in all other cases, e.g. regardless of whether any API/SSR/DSG functions are needed?

Reply:
It's enabled manually, via `GATSBY_CLOUD_IMAGE_CDN`. The check is inside the function, so we want to always run it.